### PR TITLE
Fix issue 414 CLC-418

### LINKED ISSUE #414: DMT: The migration prechecks that trigger a 'FAILED' status in the migration cluster, don't get reflected in the frontend [CLC-418]
Given a migration w/ status: 'FAILED' due to some pre-check (i.e. no IMap in source that's listed in migration config), a warning log is written from the migration cluster, but no error is shown in DMT frontend and it does not exit (spins on step '2/2' indefinitely). 

To reproduce:

Follow the README instruction included in the DMT package, but do not add any data structure to the source cluster. That is:

1. `docker run -p 127.0.0.1:5701:5701 -e HZ_CLUSTERNAME=source hazelcast/hazelcast:4.2.7`
2. `docker run -e HZ_NETWORK_PORT_PORT=5901 -p 127.0.0.1:5901:5901 -e HZ_LICENSEKEY="yourlicensekey" -e HZ_CLUSTERNAME=target hazelcast/hazelcast-enterprise:5.3.2-SNAPSHOT-slim`
3. `HZ_NETWORK_PORT_PORT=5702 HZ_CLUSTERNAME=migration ./bin/hz start`
4. `./bin/dmt_[platform]_[arch] --config migration.yaml start migration_config --yes`

### PR DESCRIPTION
**Pull request by srknzl**
fixes #414 

Adds the logic of throwing the error if migration becomes failed in WaitForMigrationToBeInProgress. 

JSON queries that are expected to return an array should use WITH WRAPPER (see https://docs.hazelcast.com/hazelcast/5.3/sql/working-with-json), I fixed that as well.

### CODE REVIEW

**Review by sumnerib — APPROVED** [review]
I cannot comment on the code itself, as I don't have much experience w/ golang, but building the tool from this branch does look to solve the issue.

File: base/commands/migration/start_stages_it_test.go [hunk 1]
```diff
@@ -12,15 +12,16 @@ import (
...
+
 	_ "github.com/hazelcast/hazelcast-commandline-client/base"
 	_ "github.com/hazelcast/hazelcast-commandline-client/base/commands"
 	"github.com/hazelcast/hazelcast-commandline-client/base/commands/migration"
 	"github.com/hazelcast/hazelcast-commandline-client/clc/paths"
 	. "github.com/hazelcast/hazelcast-commandline-client/internal/check"
 	"github.com/hazelcast/hazelcast-commandline-client/internal/it"
-	hz "github.com/hazelcast/hazelcast-go-client"
-	"github.com/hazelcast/hazelcast-go-client/serialization"
-	"github.com/stretchr/testify/require"
```
Comment (srknzl) [inline base/commands/migration/start_stages_it_test.go:23]: this is done by goland automatically and I can't prevent it doing that :D

File: base/commands/migration/migration_stages.go [hunk 2]
```diff
@@ -188,6 +188,14 @@ func WaitForMigrationToBeInProgress(ctx context.Context, ci *hazelcast.ClientInt
 			}
 			return err
 		}
+		if Status(status) == StatusFailed {
+			errs, err := fetchMigrationErrors(ctx, ci, migrationID)
+			if err != nil {
+				return fmt.Errorf("migration failed and dmt cannot fetch migration errors: %w", err)
+			} else {
```
Comment (kutluhanmetin) [inline base/commands/migration/migration_stages.go:195]: `else` is not necessary.

File: base/commands/migration/migration_stages.go [hunk 3]
```diff
@@ -235,7 +243,7 @@ func fetchMigrationReport(ctx context.Context, ci *hazelcast.ClientInternal, mig
 }
 
 func fetchMigrationErrors(ctx context.Context, ci *hazelcast.ClientInternal, migrationID string) (string, error) {
-	q := fmt.Sprintf(`SELECT JSON_QUERY(this, '$.errors') FROM %s WHERE __key='%s'`, StatusMapName, migrationID)
+	q := fmt.Sprintf(`SELECT JSON_QUERY(this, '$.errors' WITH WRAPPER) FROM %s WHERE __key='%s'`, StatusMapName, migrationID)
```
Comment (kutluhanmetin) [inline base/commands/migration/migration_stages.go:246]: What is `WITH WRAPPER` ?

Comment (frant-hartm) [reply base/commands/migration/migration_stages.go:246]: See https://docs.hazelcast.com/hazelcast/5.3/sql/functions-and-operators#json_query

File: base/commands/migration/start_stages_it_test.go [hunk 4]
```diff
@@ -42,7 +43,7 @@ func TestMigrationStages(t *testing.T) {
 				"testdata/start/migration_success_initial.json",
 				"testdata/start/migration_success_failure.json",
 			},
-			expectedErr: errors.New("Failed migrating IMAP: imap5: some error"),
+			expectedErr: errors.New("Failed migrating IMAP: imap5: [\"some error\"]"),
```
Comment (yuce) [inline base/commands/migration/start_stages_it_test.go:46]: I think this error message is not optimal.
Can't we keep the original string?

**Review by yuce — CHANGES_REQUESTED** [review]

File: base/commands/migration/start_stages_it_test.go (hunk 4 above)
Comment (kutluhanmetin) [reply base/commands/migration/start_stages_it_test.go:46]: I would rewrite `fetchMigrationErrors` function like:
```
func fetchMigrationErrors(ctx context.Context, ci *hazelcast.ClientInternal, migrationID string) (string, error) {
	q := fmt.Sprintf(`SELECT JSON_QUERY(this, '$.errors' WITH WRAPPER) FROM %s WHERE __key='%s'`, StatusMapName, migrationID)
	row, err := querySingleRow(ctx, ci, q)
	if err != nil {
		return "", err
	}
	// ex. converts ["some error", "another error"] -> some error\nanother error
	r := strings.Split(
		strings.ReplaceAll(
			strings.ReplaceAll(
				strings.ReplaceAll(string(row.(serialization.JSON)), `"`, ``),
				"[", ""),
			"]", ""),
		",")
	return strings.Join(r, "\n"), nil
}
```

Comment (kutluhanmetin) [reply base/commands/migration/start_stages_it_test.go:46]: If it is possible to iterate `errors` it would be more convenient of course.

**Review by yuce — APPROVED** [review]
Looks good!

**Review by kutluhanmetin — APPROVED** [review]
Thank you Serkan!

**Pull request merged**